### PR TITLE
Dev tools: add content on YAML tab

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Proposed change

Dev tools: add content on YAML tab
- minor cosmetic changes to headers and markdown tweaks

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/tools/dev-tools.markdown [hunk 1]
```diff
@@ -24,7 +24,26 @@ Screenshot of Home Assistant's developer tools.
 
 The Developer Tools is meant for **all** (not just for the developers) to quickly try out things - like calling services, updating states, raising events, and publishing messages in MQTT). It is also a necessary tool for those who write custom automations and scripts by hand. The following describes each of the sections in detail.
 
-## States
+## YAML tab
+
+The YAML tab provides buttons to trigger a check of specific configuration files and to reload the configuration. Reloading is needed to apply changes that you've made to the configuration.
```
Comment (ludeeus) [inline source/_docs/tools/dev-tools.markdown]:
```suggestion
The YAML tab provides buttons to trigger a check of configuration files and to reload the configuration. Reloading is needed to apply changes that you've made to the configuration.
```

File: source/_docs/tools/dev-tools.markdown [hunk 2]
```diff
@@ -24,7 +24,26 @@ Screenshot of Home Assistant's developer tools.
...
+
+The YAML tab provides buttons to trigger a check of configuration files and to reload the configuration. Reloading is needed to apply changes that you've made to the configuration.
+
+It is almost the same as the option under **Settings** > three dot menu (top right) > **Restart Home Assistant** > **Quick reload**. The only difference is that **Quick reload** reloads all the configuration, whereas this YAML tab allows you to only reload one specific configuration at a time.
+
+### Reloading the YAML configuration
+
+For configuration changes to become effective, the configuration must be reloaded. Most integrations in Home Assistant (that do not interact with {% term devices %} or {% term services %}) can reload changes made to their configuration in `configuration.yaml` without needing to restart Home Assistant.
+
+1. Go to {% my server_controls title="**Developer Tools** > **YAML**" %} and scroll down to the YAML configuration reloading section (alternatively, hit "c" anywhere in the UI and search for "reload").
```
Comment (ludeeus) [inline source/_docs/tools/dev-tools.markdown]:
```suggestion
1. Go to {% my server_controls title="**Developer Tools** > **YAML**" %} and scroll down to the YAML configuration reloading section (alternatively, hit ["c"](/docs/tools/quick-bar/) anywhere in the UI and search for "reload").
```

**Review by ludeeus — APPROVED** [review]
LGTM 👍

**Review by klaasnicolaas — APPROVED** [review]

**Comment by c0ffeeca7** [comment]
Thanks!

**Pull request merged**